pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -216,7 +217,8 @@ func (l *listUsersQuery) expandDirect(
 		return err
 	}
 
-	iter, err := l.ds.Read(ctx, req.GetStoreId(), &openfgav1.TupleKey{
+	ds := storagewrappers.NewCombinedTupleReader(l.ds, req.GetContextualTuples().GetTupleKeys())
+	iter, err := ds.Read(ctx, req.GetStoreId(), &openfgav1.TupleKey{
```

Comment:
note (non-blocking): @willvedd this needs to be a filtered read, otherwise the `FilteredTupleKeyIterator` below will be doing a lot of filtering in memory.

This doesn't need to be done in this PR, but before we close out this work we need a filtered read that only returns all possible related users/subjects that match the target subject types we need to expand. Otherwise we'll be filtering tons of shit out. This is particularly important because we can also have multi-model stores that leave residual tuples in the system that may have to get filtered out. We should strive as much as possible to avoid the in-memory filtering below this line of code.

Reply:
Good call. Won't do here but created a ticket to capture that task.